### PR TITLE
feat: add showCheckedStrategy

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/34193

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      set `showCheckedStrategy `  to `'parent'\|'child''` for value type       |
| 🇨🇳 Chinese |   设置`showCheckedStrategy `为`'parent'\|'child'`选择返回值方式      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: src/utils/treeUtil.ts [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
   const valueSet = new Set(pathKeys);
   const keyPathEntities = getKeyPathEntities();
 
+  if (showCheckedStrategy === 'child') {
+    return pathKeys.filter(key => {
+      const entity = keyPathEntities[key];
+      const children = entity ? entity.children : null;
+      if (children && children.find(child => child.key && valueSet.has(child.key))) {
+        return false;
+      }
```
Comment (MadCcc) [inline src/utils/treeUtil.ts:23]: 用 some 比较好，find 会走完整个 array

Comment (MadCcc) [reply src/utils/treeUtil.ts:23]: 另外感觉可以简化一下

File: src/Cascader.tsx [hunk 2]
```diff
@@ -43,6 +43,7 @@ export interface InternalFieldNames extends Required<FieldNames> {
 export type SingleValueType = (string | number)[];
 
 export type ValueType = SingleValueType | SingleValueType[];
+export type ShowCheckedStrategy = 'parent' | 'child';
```
Comment (MadCcc) [inline src/Cascader.tsx:46]: 像 TreeSelect 一样搞个常量？字面量不是很好维护

Comment (heiyu4585) [reply src/Cascader.tsx:46]: TreeSelect?可以帮忙具体一点位置吗 我参考下

File: src/utils/treeUtil.ts (hunk 1 above)
Comment (heiyu4585) [reply src/utils/treeUtil.ts:23]: done

File: src/Cascader.tsx (hunk 2 above)
Comment (heiyu4585) [reply src/Cascader.tsx:46]: done

**Comment by MadCcc** [comment]
`rc-cascader` v3.3.0 发布完了，后续在 antd master 分支上进行吧。

**Comment by heiyu4585** [comment]
> `rc-cascader` v3.3.0 发布完了，后续在 antd master 分支上进行吧。

不需要我做什么操作了吧

**Comment by MadCcc** [comment]
需要更新下文档和依赖，最好能和 TreeSelect 一样把 strategy 挂在 Cascader 下面

**Pull request merged**